doc/sailfish-webview-webengine.qdoc

```diff
@@ -0,0 +1,327 @@
+          Logger.debug("JSComp: ExampleComponent.js loaded");
+        }
+
+        ExampleComponent.prototype = {
+          classID: Components.ID("{20227a22-1722-4753-b8f6-c842b401b4c3}"),
+
+          QueryInterface: XPCOMUtils.generateQI([Ci.nsIObserver]),
+
+          observe: function(aSubject, aTopic, aData) {
+            switch (aTopic) {
+            case "app-startup":
+              Services.obs.addObserver(this, "exampleTopic", false);
+              break;
+            case "exampleTopic":
+              Logger.debug("ExampleComponent: exapleTopic data: " + aData);
+              break;
+            }
+          },
+        };
+
+        this.NSGetFactory = XPCOMUtils.generateNSGetFactory([ExampleComponent]);
+    \endcode
+*/
+
+/*!
+    \qmlmethod WebEngine::addObserver(aTopic)
+    \brief Registers the WebEngine for receiving notifications on a topic
+
+    The gecko and EmbedLite components use notifications to transfer data and
+    information about events to other components. The \c addObserver method
+    allows the WebEngine to be registered as interested in the particular \a
+    aTopic so that it will start receiving notifications about them.
+
+    When WebEngine receivs a notification on a registered \a aTopic it will emit
+    a \c recvObserve signal.
+
+    The Mozilla \l {https://developer.mozilla.org/en-US/docs/Mozilla/Tech/XPCOM/Reference/Interface/nsIObserverService}{nsIObserver}
+    documentation provides details about the underlying processes.
+
+    There is also a non-exhaustive list of \l {https://developer.mozilla.org/en-US/docs/Mozilla/Tech/XPCOM/Observer_Notifications}{observer notification topics}
+    that can be subscribed to.
+
+    \sa removeObserver, recvObserve, notifyObservers
+*/
+
+/*!
+    \qmlmethod WebEngine::removeObserver(aTopic)
+    \brief Unregisters the WebEngine from receiving notifications on a topic.
+
+    If a component which previously registered an interest in receiving
+    notifications on \a aTopic by calling \l addObserver is no longer interested
+    in receiving them, it can call \c removeObserver to indicate this.
+
+    Calling \c removeObserver does not guarantee that no more notifications on
+    the topic will be received, since other components may have registered an
+    interest with the \c WebEngine. The \c WebEngine will stop emitting signals
+    on the topic only once all interests have been deregistered.
+
+    \sa addObserver, recvObserve, notifyObservers
+*/
+
+/*!
+    \qmlmethod WebEngine::notifyObservers(topic, value)
+    \brief Sends a broadcast notification that can be observed by other components.
+
+    Sends a notification that will be picked up by any component that has
+    registered an observer for \a topic.
+
+    Additional data can be sent with the notification using the \a value
+    parameter in JSON format.
+
+    \sa addObserver, recvObserve
+*/
+
+/*!
+    \internal
+    \qmlmethod WebEngine::runEmbedding(aDelay)
```

Comment:
This is on of those methods that I'd have liked to see in WebEngine C++ doc side which I referred here https://github.com/sailfishos/sailfish-components-webview/pull/108/files#r639432191. Surely, not good for QML side.